src/View/Form/ContextInterface.php

```diff
@@ -48,9 +48,14 @@ public function isCreate();
      *
      * @param string $field A dot separated path to the field a value
      *   is needed for.
+     * @param array $options Options:
+     *   - `default`: Default value to return if no value found in request
+     *     data or context record.
+     *   - `schemaDefault`: Boolen indicating whether default value from
+     *      context's schema should be used if it's not explicitly provided.
      * @return mixed
      */
-    public function val($field);
+    public function val($field, $options = []);
```

Comment:
Line [2416](https://github.com/cakephp/cakephp/pull/8735/files#diff-86d5537968f770710b98fddb2840fcdaR2416) in `FormHelper` will ensure that setting `default` key still works as it currently does, even if a context class ignores `$options`.